components/Conversation/MessagesList.tsx

```diff
@@ -96,7 +96,10 @@ const MessagesList = ({
               const dateHasChanged = !isOnSameDay(lastMessageDate, msg.sent)
               lastMessageDate = msg.sent
               return dateHasChanged
-                ? [<DateDivider date={msg.sent} key={msg.id} />, tile]
+                ? [
```

Comment:
So what it wants is for the elements within a list to all have a different `key` attribute, and I think currently some elements will have msg id and some (the date dividers) will have msg sent time as key. So you're probably not seeing warnings if none of the timestamps are the same, but it's a little brittle - how about wrapping the whole returned thing in a `<div key={msg.id}>` something like this:
```js
            {messages?.map((msg: Message) => {
              const isSender = msg.senderAddress === walletAddress
              const tile = (
                <MessageTile message={msg} key={msg.id} isSender={isSender} />
              )
              const dateHasChanged = !isOnSameDay(lastMessageDate, msg.sent)
              lastMessageDate = msg.sent
              return dateHasChanged
                ? <div key={msg.id}>
                    <DateDivider date={msg.sent} key={msg.sent?.toString()} />
                    {tile}
                  </div>
                : tile
            })}
```

Reply:
@snormore Ooh that's much nicer, thank you. With the tile and date divider elements now wrapped in `<div key={msg.id}>`, do they even need keys anymore?

Reply:
The date divider doesn't need the key but since you're rendering the tile on it's own in the other half of the condition it would need the key - it shouldn't be a problem/warning or anything to leave it on the tile and have it render inside the div with the key in the other half of the condition too